Source/WebCore/platform/calc/CalcOperator.h

```diff
@@ -63,4 +65,239 @@ enum class CalcOperator : uint8_t {
+    }
+    case CalcOperator::Exp: {
+        if (children.size() != 1)
+            return std::numeric_limits<double>::quiet_NaN();
+        return std::exp(evaluate(children[0]));
+    }
+    case CalcOperator::Asin: {
+        if (children.size() != 1)
+            return std::numeric_limits<double>::quiet_NaN();
+        return rad2deg(std::asin(evaluate(children[0])));
+    }
+    case CalcOperator::Acos: {
+        if (children.size() != 1)
+            return std::numeric_limits<double>::quiet_NaN();
+        return rad2deg(std::acos(evaluate(children[0])));
+    }
+    case CalcOperator::Atan: {
+        if (children.size() != 1)
+            return std::numeric_limits<double>::quiet_NaN();
+        return rad2deg(std::atan(evaluate(children[0])));
+    }
+    case CalcOperator::Atan2: {
+        if (children.size() != 2)
+            return std::numeric_limits<double>::quiet_NaN();
+        return rad2deg(atan2(evaluate(children[0]), evaluate(children[1])));
+    }
+    case CalcOperator::Abs: {
+        if (children.size() != 1)
+            return std::numeric_limits<double>::quiet_NaN();
+        return std::abs(evaluate(children[0]));
+    }
+    case CalcOperator::Sign: {
+        if (children.size() != 1)
+            return std::numeric_limits<double>::quiet_NaN();
+        auto evaluatedChild = evaluate(children[0]);
+        if (evaluatedChild > 0)
+            return 1;
+        if (evaluatedChild < 0)
+            return -1;
+        return evaluatedChild;
+    }
+    case CalcOperator::Mod: {
+        if (children.size() != 2)
+            return std::numeric_limits<double>::quiet_NaN();
+        float left = evaluate(children[0]);
+        float right = evaluate(children[1]);
+        if (!right)
+            return std::numeric_limits<double>::quiet_NaN();
+        if ((left < 0) == (right < 0))
+            return std::fmod(left, right);
+        return std::remainder(left, right);
+    }
+    case CalcOperator::Rem: {
+        if (children.size() != 2)
+            return std::numeric_limits<double>::quiet_NaN();
+        float left = evaluate(children[0]);
+        float right = evaluate(children[1]);
+        if (!right)
+            return std::numeric_limits<double>::quiet_NaN();
+        return std::fmod(left, right);
+    }
+    case CalcOperator::Round:
+        return std::numeric_limits<double>::quiet_NaN();
+    case CalcOperator::Up: {
+        if (children.size() != 2)
+            return std::numeric_limits<double>::quiet_NaN();
+        auto valueToRound = evaluate(children[0]);
+        auto roundingInterval = evaluate(children[1]);
+        if (!std::isinf(valueToRound) && std::isinf(roundingInterval)) {
+            if (!valueToRound)
+                return valueToRound;
+            return std::signbit(valueToRound) ? -0.0 : std::numeric_limits<double>::infinity();
+        }
+        auto ret = getNearestMultiples(valueToRound, roundingInterval);
+        return ret.second;
```

Comment:
A local named `ret` is not WebKit coding style. In this case I would suggest merging into a single line. In other cases a name might be better.

Reply:
Fixed